aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java

```diff
@@ -134,22 +125,12 @@ protected String region() {
     return region;
   }
 
-  protected String s3Endpoint() {
-    return s3Endpoint;
-  }
-
-  protected String httpClientType() {
-    return httpClientType;
-  }
-
-  protected boolean s3UseArnRegionEnabled() {
-    return s3UseArnRegionEnabled;
+  protected AwsProperties awsProperties() {
+    return awsProperties;
   }
 
   private StsClient sts() {
-    return StsClient.builder()
-        .httpClientBuilder(AwsClientFactories.configureHttpClientBuilder(httpClientType))
-        .build();
+    return StsClient.builder().applyMutation(awsProperties::applyHttpClientConfiguration).build();
```

Comment:
nit: newline for each part of chain in the builder

```
tsClient.builder()
    .applyMutation(awsProperties::applyHttpClientConfiguration)
    .build()
```

Reply:
It seems that the current format of `line 133` is favored by `spotlessApply`. I tried to make new line for each part and triggered the `spotlessCheck` violation. 

I did some additional tests and found that we need at least two consecutive configurations so that `spotlessCheck` allow us to make new line for each part:
```
tsClient.builder()
    .applyMutation(awsProperties::applyHttpClientConfiguration)
    .applyMutation(<any configuration>)
    .build()
```
is allowed. But if we only have one `applyMutation`, it will be auto-formated to
```
tsClient.builder().applyMutation(awsProperties::applyHttpClientConfiguration).build()
```